### PR TITLE
Add logstash to stack as an optional service

### PR DESCRIPTION
**Pull request by bhapas**
Fixes - https://github.com/elastic/elastic-package/issues/1432

### Screenshot for Doc changes

image

### CODE REVIEW

File: internal/install/application_configuration.go [hunk 1]
```diff
@@ -87,6 +88,7 @@ func (s stack) ImageRefOverridesForVersion(version string) ImageRefs {
 		ElasticAgent:  checkImageRefOverride("ELASTIC_AGENT_IMAGE_REF_OVERRIDE", stringOrDefault(appConfigImageRefs.ElasticAgent, "")),
 		Elasticsearch: checkImageRefOverride("ELASTICSEARCH_IMAGE_REF_OVERRIDE", stringOrDefault(appConfigImageRefs.Elasticsearch, "")),
 		Kibana:        checkImageRefOverride("KIBANA_IMAGE_REF_OVERRIDE", stringOrDefault(appConfigImageRefs.Kibana, "")),
+		Logstash:      checkImageRefOverride("LOGSTAHS_IMAGE_REF_OVERRIDE", stringOrDefault(appConfigImageRefs.Logstash, "")),
```
Comment (marc-gr) [inline internal/install/application_configuration.go:91]:
```suggestion
		Logstash:      checkImageRefOverride("LOGSTASH_IMAGE_REF_OVERRIDE", stringOrDefault(appConfigImageRefs.Logstash, "")),
```

**Review by mrodm — COMMENTED** [review]
Thanks @bhapas for this PR!
I've added some suggestions related to docs and the setting up of the kibana policy.

I've tested locally and it looks great!

File: internal/testrunner/runners/system/runner.go [hunk 2]
```diff
@@ -516,10 +516,20 @@ func (r *runner) runTest(config *testConfig, ctxt servicedeployer.ServiceContext
...
+
 	p := kibana.Policy{
-		Name:        fmt.Sprintf("ep-test-system-%s-%s-%s", r.options.TestFolder.Package, r.options.TestFolder.DataStream, testTime),
-		Description: fmt.Sprintf("test policy created by elastic-package test system for data stream %s/%s", r.options.TestFolder.Package, r.options.TestFolder.DataStream),
-		Namespace:   "ep",
+		Name:         fmt.Sprintf("ep-test-system-%s-%s-%s", r.options.TestFolder.Package, r.options.TestFolder.DataStream, testTime),
+		Description:  fmt.Sprintf("test policy created by elastic-package test system for data stream %s/%s", r.options.TestFolder.Package, r.options.TestFolder.DataStream),
+		Namespace:    "ep",
+		DataOutputID: elasticAgentDataOutputId,
 	}
```
[the remark this answers is not in the document]
Comment (bhapas) [reply internal/testrunner/runners/system/runner.go:533]: Agree. The other runners do not need to care about the DataOutputId as of now

**Review by mrodm — APPROVED** [review]
Thanks @bhapas !

**Pull request merged**